File: mobile-app/src/Types/response/NotificationResponse.ts
import FileResponse from "./FileResponse";

export enum NotificationType {
  COMMENT = "COMMENT",
  LIKE = "LIKE",
  MESSAGE = "MESSAGE",
}

export interface NotificationActorResponse {
  id: string;
  displayName: string;
  shownName: string;
  avatar: FileResponse | null;
}

export interface NotificationResponse {
  id: string;
  notificationType: NotificationType;
  title: string;
  body: string;
  previewText: string | null;
  feedItemId: string | null;
  commentId: string | null;
  conversationId: string | null;
  messageId: string | null;
  read: boolean;
  createdAt: string; // ISO string
  readAt: string | null; // ISO string
  actor: NotificationActorResponse | null;
}

File: backend/src/main/java/com/hehe/thesocial/dto/response/notification/NotificationResponse.java
package com.hehe.thesocial.dto.response.notification;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.hehe.thesocial.entity.enums.NotificationType;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.FieldDefaults;

import java.time.Instant;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
@JsonInclude(JsonInclude.Include.NON_NULL)
public class NotificationResponse {
    String id;
    NotificationType notificationType;
    String title;
    String body;
    String previewText;
    String feedItemId;
    String commentId;
    String conversationId;
    String messageId;
    boolean read;
    Instant createdAt;
    Instant readAt;
    NotificationActorResponse actor;
}








